tests/toranj/openthread-core-toranj-config.h

```diff
@@ -169,6 +169,8 @@
 
 #define OPENTHREAD_CONFIG_DNS_CLIENT_ENABLE 1
 
+#define OPENTHREAD_CONFIG_DNS_CLIENT_DEFAULT_SERVICE_MODE OT_DNS_SERVICE_MODE_SRV_TXT_OPTIMIZE
```

Comment:
Is there a reason to change the default before running these tests? Of course it's good to test the "optimize" mode, but is the "separate" mode then also tested somewhere?

Reply:
Yes. All combinations of "service modes" and server behaviors are tested and covered in the unit test `test_dns_client.hpp`.
https://github.com/openthread/openthread/blob/4e8f3c06c1cce216070d821cca91b38be150d2c8/tests/unit/test_dns_client.cpp#L629-L901